common/makefile.mk

```diff
@@ -55,21 +44,41 @@ test: $(JUJU_VERSION)
 endif
 
 .PHONY: install
-install: $(JUJU_VERSION)
-	install -D fake-jujud $(JUJU_INSTALLED)
+install:
+	install -D $(SERVER) $(DESTDIR)/usr/bin/$(SERVER)-$(JUJU_VERSION)
+	install -D fake-juju $(DESTDIR)/usr/bin/$(CLIENT)-$(JUJU_VERSION)
 
-.PHONY: clean
-clean:
-	rm -f $(JUJU_TARBALL)
-	rm -rf $(JUJU_SRC)
-	rm -f $(JUJU_VERSION)
+# Copy fake-juju Go code extensions to the upstream source tree
+.PHONY: overlay
+overlay: src
+	# First clean up the upstream source tree from any overlay symlink
+	# we might have created in previous runs.
+	find src/ -name *-fakejuju.go | xargs rm -f
 
+	# Then create new symlinks against our overlay Go source files.
+	for name in $(shell find ../common/core/ -name "*.go"); \
+		do ln -s $$(pwd)/$$name src/$$(echo $$name | cut -d / -f 4-); \
```

Comment:
Essentially it's stripping the first 3 segments of a non-absolute path:

```
~$ echo ../common/core/github.com/juju/testing/mgo-fakejuju.go | cut -d / -f 4-
github.com/juju/testing/mgo-fakejuju.go
```

Perhaps I could replace it with a sed if that's more clear:

```
~$ echo ../common/core/github.com/juju/testing/mgo-fakejuju.go | sed -e "s|^../common/core/||"
github.com/juju/testing/mgo-fakejuju.go
```